Gemfile

```diff
@@ -65,6 +65,7 @@ group :test, :development do
   gem 'rack-test', :require => 'rack/test'
   gem 'simplecov', :require => false
   gem 'pry-byebug'
+  gem 'pry-doc'
```

Comment:
Well I guess I just haven't got the right balance between separation of concerns and `git rebase` to consolidate or simplify the history.  There's a ton of commits that have been consolidated into this PR.
Anyhow, I get the point and I'll keep it in mind for future PR.  I have bigger fish to fry rather than figure out how to fix this one for this PR.